libraries/pushproviders/unifiedpush/src/main/kotlin/io/element/android/libraries/pushproviders/unifiedpush/UnifiedPushRemovedGatewayHandler.kt

```diff
@@ -58,14 +80,21 @@ class DefaultUnifiedPushRemovedGatewayHandler(
                         val distributor = pushProvider?.getCurrentDistributor(userId)
                         // Attempt to register again
                         if (pushProvider != null && distributor != null) {
-                            pushService.registerWith(
-                                client,
-                                pushProvider,
-                                distributor,
-                            )
-                                .onFailure {
-                                    Timber.tag(loggerTag.value).w(it, "Unable to register with current data")
-                                }
+                            if (unifiedPushRemovedGatewayThrottler.canRegisterAgain()) {
+                                pushService.registerWith(
```

Comment:
Maybe add some comments for the reason this is needed?

Reply:
Sure, you mean explain why `canRegisterAgain()` is needed?

Reply:
Ah that's not really that. We do not automatically retry in case of failure.
See my second video in the description. We avoid registering again it the user persist to delete the topic in th ntfy application.